server/src/main/java/com/cloud/storage/VolumeApiServiceImpl.java

```diff
@@ -3007,6 +3009,13 @@ private int getMaxDataVolumesSupported(UserVmVO vm) {
         if (host != null) {
             _hostDao.loadDetails(host);
             maxDataVolumesSupported = _hypervisorCapabilitiesDao.getMaxDataVolumesLimit(host.getHypervisorType(), host.getDetail("product_version"));
+        } else {
+            List<HypervisorType> supportingDefaultHV = Arrays.asList(HypervisorType.XenServer, HypervisorType.VMware,
+                    HypervisorType.KVM, HypervisorType.Ovm, HypervisorType.LXC);
+            HypervisorType hypervisorType = vm.getHypervisorType();
+            if (hypervisorType != null && supportingDefaultHV.contains(hypervisorType)) {
+                maxDataVolumesSupported = _hypervisorCapabilitiesDao.getMaxDataVolumesLimit(hypervisorType, "default");
```

Comment:
Could version specific values cause regression or bug?

Reply:
if these are lower the "default", yes @rhtyd. But that should never be the case, should it?

Reply:
Good point, I've verified any version contains max limit lower than the default for each hypervisor